source/magnifier/magnifier.py

```diff
@@ -0,0 +1,297 @@
+
+	@lastMousePosition.setter
+	def lastMousePosition(self, value: tuple[int, int]) -> None:
+		self._lastMousePosition = value
+
+	@property
+	def lastScreenPosition(self) -> tuple[int, int]:
+		return self._lastScreenPosition
+
+	@lastScreenPosition.setter
+	def lastScreenPosition(self, value: tuple[int, int]) -> None:
+		self._lastScreenPosition = value
+
+	@property
+	def currentCoordinates(self) -> tuple[int, int]:
+		return self._currentCoordinates
+
+	@currentCoordinates.setter
+	def currentCoordinates(self, value: tuple[int, int]) -> None:
+		self._currentCoordinates = value
+
+	@property
+	def timer(self) -> wx.Timer | None:
+		return self._timer
+
+	@timer.setter
+	def timer(self, value: wx.Timer | None) -> None:
+		self._timer = value
+
+	@property
+	def filter(self) -> filter:
+		return self._filter
+
+	@filter.setter
+	def filter(self, value: filter) -> None:
+		self._filter = value
+
+	# Functions
+
+	def _startMagnifier(self) -> None:
+		"""
+		Start the magnifier.
+		"""
+		if self.isActive:
+			return
+		self.isActive = True
+		self.currentCoordinates = self._getFocusCoordinates()
+
+	def _updateMagnifier(self) -> None:
+		"""
+		Update the magnifier position and content.
+		"""
+		if not self.isActive:
+			return
+		self.currentCoordinates = self._getFocusCoordinates()
+		self._doUpdate()
+		self._startTimer(self._updateMagnifier)
+
+	def _doUpdate(self) -> None:
+		"""
+		Perform the actual update of the magnifier.
+		"""
+		raise NotImplementedError("Subclasses must implement this method.")
+
+	def _stopMagnifier(self) -> None:
+		"""
+		Stop the magnifier.
+		"""
+		if not self.isActive:
+			return
+		self._stopTimer()
+		self.isActive = False
+
+	def _zoom(self, direction: bool) -> None:
+		"""
+		Adjust the zoom level of the magnifier.
+
+		:param direction: True to zoom in, False to zoom out.
+		"""
+		if direction:
+			self.zoomLevel += self._ZOOM_STEP
+		else:
+			self.zoomLevel -= self._ZOOM_STEP
+
+	def _startTimer(self, callback: Callable[[], None] = None) -> None:
+		"""
+		Start the timer with a callback function.
+
+		:param callback: The function to call when the timer expires.
+		"""
+		self._stopTimer()
+		self.timer = wx.Timer()
+		self.timer.Bind(wx.EVT_TIMER, lambda evt: callback())
+		self.timer.Start(self._TIMER_INTERVAL_MS, oneShot=True)
+
+	def _stopTimer(self) -> None:
+		"""
+		Stop timer execution.
+		"""
+		if self.timer:
+			if self.timer.IsRunning():
+				self.timer.Stop()
+			self.timer = None
+		else:
+			log.info("no timer to stop")
+
+	def _getMagnifierPosition(self, x: int, y: int) -> tuple[int, int, int, int]:
+		"""
+		Compute the top-left corner of the magnifier window centered on (x, y).
+
+		:param x: Focus x
+		:param y: Focus y
+		:param targetWidth: Target width (defaults to screen width for fullscreen)
+		:param targetHeight: Target height (defaults to screen height for fullscreen)
+		Returns:
+			left, top, visibleWidth, visibleHeight: The position and size of the magnifier window.
```

Comment:
I suggest turning this into a dataclass or NamedTuple

Reply:
Instead of returning an unnamed tuple, create a data structure like a [NamedTuple](https://typing.python.org/en/latest/spec/namedtuples.html#defining-named-tuples) or [dataclass](https://docs.python.org/3/library/dataclasses.html#module-dataclasses) to group these variables